src/Analyzers/Core/Analyzers/MakeFieldReadonly/MakeFieldReadonlyDiagnosticAnalyzer.cs

```diff
@@ -102,19 +103,29 @@ void OnSymbolEnd(SymbolAnalysisContext symbolEndContext)
                     }
                 }
 
-                static bool IsCandidateField(IFieldSymbol symbol) =>
+                static bool IsCandidateField(IFieldSymbol symbol, Compilation compilation) =>
                         symbol.DeclaredAccessibility == Accessibility.Private &&
                         !symbol.IsReadOnly &&
                         !symbol.IsConst &&
                         !symbol.IsImplicitlyDeclared &&
                         symbol.Locations.Length == 1 &&
                         symbol.Type.IsMutableValueType() == false &&
-                        !symbol.IsFixedSizeBuffer;
+                        !symbol.IsFixedSizeBuffer &&
+                        !IsDataContractSerializable(symbol, compilation);
+
+                static bool IsDataContractSerializable(IFieldSymbol symbol, Compilation compilation)
+                {
+                    var dataMemberAttribute = compilation.DataMemberAttribute();
+                    var dataContractAttribute = compilation.DataContractAttribute();
```

Comment:
❗  Since this method is part of the analyzer fast path, these lookups should only occur once (within the Compilation Start callback)

Reply:
➡️ Now fixed